### PR TITLE
Add @types/jest-cli

### PR DESCRIPTION
**Pull request by lifeiscontent**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by lifeiscontent** [comment]
this is a WIP. I'll finish it soon.

File: types/jest-cli/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+declare module "jest-cli" {
```
Comment (andy-ms) [inline types/jest-cli/index.d.ts]: You don't need `declare module`,  just write the exports directly (as in `export type Argv = string[];` with no wrapper)

**Review by ghost — CHANGES_REQUESTED** [review]
There is a guide to [creating a new package](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package). Note this is currently missing some required files. The linter (`npm run lint jest-cli`) will tell you about other things.

File: types/jest-cli/index.d.ts (hunk 1 above)
Comment (lifeiscontent) [reply types/jest-cli/index.d.ts]: @andy-ms is that for all things?

Comment (andy-ms) [reply types/jest-cli/index.d.ts]: Yes, export all the things

File: types/jest-cli/jest-tests.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import jest from "jest-cli";
```
Comment (andy-ms) [inline types/jest-cli/jest-tests.ts]: You probably meant `import * as jest from "jest-cli";`.

File: types/jest-cli/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": ["es6", "dom"],
+        "noImplicitAny": true,
+        "noImplicitThis": false,
```
Comment (andy-ms) [inline types/jest-cli/tsconfig.json]: Should be `true` for all new packages.

File: types/jest-cli/jest-tests.ts (hunk 2 above)
Comment (lifeiscontent) [reply types/jest-cli/jest-tests.ts]: @andy-ms is that the proper import functionality?

Comment (andy-ms) [reply types/jest-cli/jest-tests.ts]: Right. `import jest from "jest-cli";` is a default import -- it gets the module's export named `default`. In this case there's no default export. `import * as jest from "jest-cli";` gets an object containing all exports.

**Pull request merged**